src/Core/Context/LanguageContext.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Context;
+
+use PrestaShop\PrestaShop\Core\Language\LanguageInterface;
+use PrestaShop\PrestaShop\Core\Localization\LocaleInterface;
+use PrestaShop\PrestaShop\Core\Localization\Specification\NumberInterface;
+
+/**
+ * This context service gives access to all contextual data related to language.
+ *
+ * It also implements some core interfaces that are used in many places in the code so that the
+ * context can be injected and used in place of these interfaces.
+ */
+class LanguageContext implements LanguageInterface, LocaleInterface
+{
+    public function __construct(
+        private readonly int $id,
+        private readonly string $name,
+        private readonly string $isoCode,
+        private readonly string $locale,
+        private readonly string $languageCode,
+        private readonly bool $isRTL,
+        private readonly string $dateFormat,
+        private readonly string $dateTimeFormat,
+        private readonly LocaleInterface $localizationLocale
+    ) {
```

Comment:
The other contexts contain protected attributes, so i think we should be consistent everywhere